### DISCUSSION TITLE
MPI ram usage

### DISCUSSION THREAD

**Discussion by Weroxig** [ggml-org/llama.cpp#6524] [Q&A]
> The above will distribute the computation across 2 processes on the first host and 1 process on the second host. Each process will use roughly an equal amount of RAM. Try to keep these numbers small, as inter-process (intra-host) communication is expensive.

When this part of the readme says RAM, does it mean the amount of ram the model uses in memory? Is this how I split how much system RAM is used? How do I configure VRAM usage then for say, 2 computers each with 1x 8GB GPU and 32GB ram, but I want system 'A' to use more ram if the model is to be offloaded to RAM (not enough vram)

**Comment by AutonomicPerfectionist** [comment]
The MPI documentation is severely out of date and doesn't mention that it doesn't even work at the moment. I have a PR with a mostly functional complete redesign that also allows you to specify how to split the model amongst the nodes. At the moment it doesn't support GPUs but I've been working on fixing that.

**Reply by Weroxig** [reply]
alright, and do you know if MPI work connecting to a machine running WSL2 instead of plain linux? 
thanks

**Reply by AutonomicPerfectionist** [reply]
I don't have a Windows box to test it but I would imagine WSL2 would work. I would also expect, however, for it to have decreased performance compared to bare-metal Linux
